platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/FelixConfig.java

```diff
@@ -0,0 +1,114 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.configuration;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+import org.osgi.service.cm.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for enabling functional constructs with {@link Configuration}s and exposes the
+ * felix.fileinstall.filename property, if it exists, as a {@link File}.
+ */
+public class FelixConfig {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FelixConfig.class);
+
+    private static final String FELIX_FILENAME_PROP = "felix.fileinstall.filename";
+
+    private final Configuration config;
+
+    private final File felixFile;
+
+    FelixConfig(Configuration config) {
+        if (config == null) {
+            throw new IllegalArgumentException("Configuration can't be null");
+        }
+        this.config = config;
+        this.felixFile = extractFelixFileInstallProp(config);
+    }
+
+    /**
+     * @return the pid for this configuration.
+     */
+    String getPid() {
+        return config.getPid();
+    }
+
+    /**
+     * @return the File object pointing to the this configuration on disk.
+     */
+    @Nullable
+    File getFile() {
+        return felixFile;
+    }
+
+    /**
+     * Determine if the felix.fileinstall.filename property changed.
+     *
+     * @param original file location to compare to the current file location poitned to by this
+     *                 config.
+     * @return False if the file locations by absolute path are identical. True otherwise.
+     */
+    boolean filePropChanged(File original) {
+        return !Objects.equals(felixFile, original);
+    }
+
+    private File extractFelixFileInstallProp(Configuration config) {
+        if (config.getProperties() == null) {
+            LOGGER.debug("Config props dictionary is null, was the config just created?");
+            return null;
+        }
+        Object felixConfigFileName = config.getProperties()
+                .get(FELIX_FILENAME_PROP);
+        if (felixConfigFileName == null) {
+            LOGGER.debug("Felix filename prop not found for pid {}", config.getPid());
+            return null;
+        }
+        return createFileFromFelixProp(felixConfigFileName);
+    }
+
+    /**
+     * Code adopted from Karaf's Config Repository Impl:
+     * https://github.com/apache/karaf/blob/master/config/src/main/java/org/apache/karaf/config/core/impl/
+     * ConfigRepositoryImpl.java#L101-L114
+     */
+    private File createFileFromFelixProp(Object felixConfigFileName) {
```

Comment:
Let's refactor this in the future. Until we have 3 separate usages for the logic, it's typically not worth the overhead.